neo3/wallet/wallet.py

```diff
@@ -0,0 +1,198 @@
+from __future__ import annotations
+
+import json
+import os.path
+from typing import Any, Dict, List, Optional
+
+from jsonschema import validate  # type: ignore
+
+from neo3.core import IJson
+from neo3.wallet.account import Account
+from neo3.wallet.scrypt_parameters import ScryptParameters
+
+
+def encode_scrypt_parameters(scrypt_parameters):
+    if isinstance(scrypt_parameters, ScryptParameters):
+        return {'n': scrypt_parameters.n, 'r': scrypt_parameters.r, 'p': scrypt_parameters.p,
+                'length': scrypt_parameters.length}
+    else:
+        type_name = scrypt_parameters.__class__.__name__
+        raise TypeError(f"Object of type '{type_name}' is not JSON serializable")
+
+
+# A sample schema, like what we'd get from json.load()
+schema = {
+    "type": "object",
+    "properties": {
+        "path": {"type": "string"},
+        "name": {"type": "string"},
+        "scrypt": {"$ref": "#/$defs/scrypt_parameters"},
+        "accounts": {
+            "type": "array",
+            "items": {"$ref": "#/$defs/account"},
+            "minItems": 0,
+        },
+        "extra": {"type": ["object", "null"],
+                  "properties": {},
+                  "additionalProperties": True
+                  },
+    },
+    "required": ["path", "name", "scrypt", "accounts", "extra"],
+    "$defs": {
+        "account": {
+            "type": "object",
+            "properties": {
+                "address": {"type": "string"},
+                "label": {"type": "string"},
+                "is_default": {"type": "boolean"},
+                "lock": {"type": "boolean"},
+                "key": {"type": "string"},
+                "contract": {"type": ""},
+                "extra": {"type": ["object", "null"],
+                          "properties": {},
+                          "additionalProperties": True}
+            },
+            "required": ["address", "label", "is_default", "lock", "key", "contract", "extra"]
+
+        },
+        "scrypt_parameters": {
+            "type": "object",
+            "properties": {
+                "n": {"type": "integer"},
+                "r": {"type": "integer"},
+                "p": {"type": "integer"},
+                "length": {"type": "integer"}
+            },
+            "required": ["n", "r", "p"]
+        }
+    }
+}
+
+
+class Wallet(IJson):
+
+    _wallet_version = '3.0'
+
+    def __init__(self,
+                 path: str,
+                 version: str,
```

Comment:
I think it's good that we can specify the `version`, however right now no version other than `3.0` is actually valid. I think we can simplify the initialiser for users by making version default initialised to `version: str = '3.0'`